tensorflow_datasets/core/download/kaggle.py

```diff
@@ -46,10 +47,44 @@
 class KaggleFile(object):
   """Represents a Kaggle competition file."""
   _URL_PREFIX = "kaggle://"
+  kaggle_type = collections.namedtuple(
+      "KaggleType",
+      ["prefix", "download_cmd", "dl_flag", "extra_flag"])
+  _DATASET_TYPE = kaggle_type(
+      prefix="dataset",
+      download_cmd="datasets",
+      dl_flag="-d",
+      extra_flag="--unzip")
+  _COMPETITION_TYPE = kaggle_type(
+      prefix="competition",
+      download_cmd="competitions",
+      dl_flag="-c",
+      extra_flag="")
+  KAGGLE_TYPES = {
+      _COMPETITION_TYPE.prefix: _COMPETITION_TYPE,
+      _DATASET_TYPE.prefix: _DATASET_TYPE
+      }
```

Comment:
It is good practice to limit the number of entry point, so the same object is only accessible from a single place at a time.
I would only keep `KAGGLE_TYPES` and remove `_COMPETITION_TYPE ` and `_DATASET_TYPE `.

Reply:
Done. :)